nomad/fsm.go

```diff
@@ -1459,54 +1474,53 @@ func (n *nomadFSM) Restore(old io.ReadCloser) error {
 			if err := dec.Decode(node); err != nil {
 				return err
 			}
-
-			// Handle upgrade paths
-			node.Canonicalize()
-
-			if err := restore.NodeRestore(node); err != nil {
-				return err
+			if filter.Include(node) {
+				node.Canonicalize() // Handle upgrade paths
+				if err := restore.NodeRestore(node); err != nil {
+					return err
+				}
 			}
 
 		case JobSnapshot:
 			job := new(structs.Job)
 			if err := dec.Decode(job); err != nil {
 				return err
 			}
-
-			/* Handle upgrade paths:
-			 * - Empty maps and slices should be treated as nil to avoid
-			 *   un-intended destructive updates in scheduler since we use
-			 *   reflect.DeepEqual. Starting Nomad 0.4.1, job submission sanitizes
-			 *   the incoming job.
-			 * - Migrate from old style upgrade stanza that used only a stagger.
-			 */
-			job.Canonicalize()
-
-			if err := restore.JobRestore(job); err != nil {
-				return err
+			if filter.Include(job) {
+				/* Handle upgrade paths:
+				 * - Empty maps and slices should be treated as nil to avoid
+				 *   un-intended destructive updates in scheduler since we use
+				 *   reflect.DeepEqual. Starting Nomad 0.4.1, job submission sanitizes
```

Comment:
```suggestion
				 *   reflect.DeepEqual
```

I've noticed `slices.Equal` and `maps.Equal` do the Right Thing; will be nice to purge `DeepEqual` soon